cep-20.2.md

```diff
@@ -0,0 +1,711 @@
+- cater to needs for automation and dependency tree analysis via a determinstic
+  format
+
+## Major differences with conda-build
+
+Outputs and top-level package are mutually exclusive, and outputs have exactly the same structure as top-level keys.
+If outputs exist, the top-level keys are 'merged' with the output keys (e.g. for the about section).
+
+# Format
+
+## Schema version
+
+The implicit version of the YAML schema for a recipe is an integer 1. 
+To discern between the "old" format and the new format, we utilize the file name.
+The old format is `meta.yaml` and the new format is `recipe.yaml`.
+The version can be explicitly set by adding a `schema_version` key to the recipe.
+
+```yaml
+# optional, since implicitly defaults to 1
+schema_version: 1  # integer
+```
+
+To benefit from autocompletion, and other LSP features in editors, we can add a schema URL to the recipe.
+
+```yaml
+# yaml-language-server: $schema=https://raw.githubusercontent.com/prefix-dev/recipe-format/main/schema.json
+```
+
+## Context section
+
+```yaml
+# The context dictionary defines arbitrary key-value pairs for Jinja interpolation
+# and replaces the {% set ... %} commands commonly used in recipes
+context:
+  variable: test
+  # note that we can reference previous values, that means that they are rendered in order
+  other_variable: test_${{ variable }}
+```
+
+## Package section
+
+```yaml
+package:
+  # The name of the package
+  name: string
+  # The version of the package, following the conda version spec
+  # Note that versions are always strings (1.23 is _not_ a valid version since it is a `float`. Needs to be quoted.)
+  version: string
+```
+
+## Build section
+
+```yaml
+build:
+  # the build number
+  number: Option<integer>, defaults to 0
+  # the build string. This is usually ommited
+  string: Option<string>, defaults to a build string made from "package hash & build number"
+
+  # A list of conditions under which to skip the build of this package
+  skip: [list of expressions]
+
+  force_use_keys: [string]
+  force_ignore_keys: [string]
+
+  # wether the package is a noarch package, and if yes, wether it is "generic" or "python"
+  noarch: Option<"generic" | "python">
+
+  # PythonEntryPoint: `bsdiff4 = bsdiff4.cli:main_bsdiff4`
+  entry_points: [PythonEntryPoint]
+
+  # Use the python.app for the entrypoint (not exactly sure what that means in practice)
+  osx_is_app: bool (default false)
+
+  # used on conda-forge, still needed?
+  preserve_egg_dir: bool (default false)
+
+  # only used as a hack to down-prioritize packages
+  track_features: [string]
+
+  # skip compiling pyc for some files
+  skip_compile_pyc: [glob]
+
+  # do not soft- or hard-link these files, but always copy them
+  no_link: [glob]
+
+  # the script that is executed to build the package
+  # if it is only one element and ends with `.sh` or `.bat`
+  script: string | [string]
+
+  # linux only, list of rpaths
+  rpaths: [path] (defaults to ['lib/'])
```

Comment:
This currently works on osx (though undocumented), but it would be really nice if we could have the same (or similar) functionality on windows, which [comes up](https://github.com/conda/conda-build/issues/4753) for example in adding SP_DIR to the rpath for some packages.